e2e/libs/recurring_job/rest.py

```diff
@@ -0,0 +1,180 @@
+
+    def check_jobs_work(self, volume_name):
+        # check if snapshot/backup is really created by the
+        # recurring job following the cron schedule
+        # currently only support checking normal snapshot/backup
+        # every 1 min recurring job
+        jobs, _ = self.get_volume_recurring_jobs_and_groups(volume_name)
+        for job_name in jobs:
+            job = self.get(job_name)
+            logging.warn(f"get recurring job: {job}")
+            if job['task'] == 'snapshot' and job['cron'] == '* * * * *':
+                period_in_sec = 60
+                self._check_snapshot_created_in_time(volume_name, job_name, period_in_sec)
+            elif job['task'] == 'backup' and job['cron'] == '* * * * *':
+                period_in_sec = 60
+                self._check_backup_created_in_time(volume_name, period_in_sec)
+
+    def _check_snapshot_created_in_time(self, volume_name, job_name, period_in_sec):
+        # check snapshot can be created by the recurring job
+        current_time = datetime.utcnow()
+        logging.warn(f"current_time = {current_time}")
+        current_timestamp = current_time.timestamp()
+        logging.warn(f"current_timestamp = {current_timestamp}")
+        label_selector=f"longhornvolume={volume_name}"
+        snapshot_timestamp = 0
+        for _ in range(period_in_sec * 2):
+            snapshot_list = filter_cr("longhorn.io", "v1beta2", "longhorn-system", "snapshots", label_selector=label_selector)
+            try:
+                if len(snapshot_list['items']) > 0:
+                    for item in snapshot_list['items']:
+                        # this snapshot can be created by snapshot or backup recurring job
+                        # but job_name is in spec.labels.RecurringJob
+                        # and crd doesn't support field selector
+                        # so need to filter by ourselves
+                        if item['spec']['labels']['RecurringJob'] == job_name:
+                            logging.warn(f"item = {item}")
+                            snapshot_time = snapshot_list['items'][0]['metadata']['creationTimestamp']
+                            logging.warn(f"snapshot_time = {snapshot_time}")
+                            snapshot_time = datetime.strptime(snapshot_time, '%Y-%m-%dT%H:%M:%SZ')
+                            logging.warn(f"snapshot_time = {snapshot_time}")
+                            snapshot_timestamp = snapshot_time.timestamp()
+                            logging.warn(f"snapshot_timestamp = {snapshot_timestamp}")
+                            break
+                    if snapshot_timestamp > current_timestamp:
+                        return
+            except Exception as e:
+                logging.warn(f"iterate snapshot list error: {e}")
+            time.sleep(1)
+        assert False, f"since {current_time},\
+                        there's no new snapshot created by recurring job \
+                        {snapshot_list}"
```

Comment:
Remember trying to use empty line to group each group of operation statements to make the code easy to read.